### PR TITLE
[d3d12] fix being able to use `dispatchWorkgroupsIndirect` without validation

### PR DESCRIPTION
**Pull request by teoxoy**
Follow-up to https://github.com/gfx-rs/wgpu/pull/5730 (which introduced the issue).

Trying to use `dispatchWorkgroupsIndirect` without the `indirect-validation` feature flag would result in:

```
Device creation failed: The parameter is incorrect. (0x80070057) 
```

### CODE REVIEW

File: wgpu-core/src/device/resource.rs [hunk 1]
```diff
@@ -2612,10 +2612,16 @@ impl Device {
             .map(|bgl| bgl.raw())
             .collect::<ArrayVec<_, { hal::MAX_BIND_GROUPS }>>();
 
+        #[cfg(feature = "indirect-validation")]
+        let additional_flags = hal::PipelineLayoutFlags::INDIRECT_BUILTIN_UPDATE;
+        #[cfg(not(feature = "indirect-validation"))]
+        let additional_flags = hal::PipelineLayoutFlags::empty();
+
```
Comment (ErichDonGubler) [inline wgpu-core/src/device/resource.rs]: suggestion(non-blocking): This seems like a great time to use `cfg!(…)`:

```suggestion
        let additional_flags = if cfg!(feature = "indirect-validation") {
            hal::PipelineLayoutFlags::INDIRECT_BUILTIN_UPDATE
        } else {
            hal::PipelineLayoutFlags::empty()
        };
```

File: wgpu-hal/src/dx12/mod.rs [hunk 2]
```diff
@@ -946,7 +946,7 @@ unsafe impl Sync for PipelineLayoutShared {}
 #[derive(Debug, Clone)]
 struct PipelineLayoutSpecialConstants {
     root_index: RootIndex,
-    cmd_signatures: CommandSignatures,
+    cmd_signatures: Option<CommandSignatures>,
```
Comment (ErichDonGubler) [inline wgpu-hal/src/dx12/mod.rs]: question: It seems like this is only used for indirect validation? If so, maybe we could rename this to communicate that better? `cmd_signatures` seems a pretty generic name, to me.

**Review by ErichDonGubler — APPROVED** [review]
As usual, LGTM, but with a question.

File: wgpu-hal/src/dx12/mod.rs (hunk 2 above)
Comment (teoxoy) [reply wgpu-hal/src/dx12/mod.rs]: Yeah, I'll rename it to `indirect_cmd_signatures`.

**Comment by teoxoy** [comment]
I don't think we need to backport this since wgpu unconditionally enables the `indirect-validation` feature which works.

**Pull request merged**